src/main/java/net/dv8tion/jda/api/events/automod/AutoModerationActionExecutionEvent.java

```diff
@@ -0,0 +1,154 @@
+import net.dv8tion.jda.api.entities.User;
+import net.dv8tion.jda.api.entities.automod.AutoModerationAction;
+import net.dv8tion.jda.api.entities.automod.TriggerType;
+import net.dv8tion.jda.api.events.Event;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+/**
+ * Indicates that an {@link net.dv8tion.jda.api.entities.AutoModerationRule AutoModerationRule} was excuted.
+ */
+public class AutoModerationActionExecutionEvent extends Event
+{
+    protected final AutoModerationRule rule;
+
+    private final AutoModerationAction executedAction;
+    private final User triggerer;
+    private final TriggerType ruleTriggerer;
+    private final GuildChannel channel;
+    private final long messageId;
+    private final long alertSystemMessageId;
+    private final String content;
+    private final String matchedKeyword;
+    private final String matchedContent;
+
+
+    public AutoModerationActionExecutionEvent(@Nonnull JDA api, long responseNumber, @Nonnull AutoModerationRule rule, @Nonnull AutoModerationAction executedAction, @Nonnull User triggerer, @Nonnull TriggerType ruleTriggerer, @Nullable GuildChannel channel, @Nullable long messageId, @Nullable long alertSystemMessageId, @Nonnull String content, @Nullable String matchedKeyword, @Nullable String matchedContent)
+    {
+        super(api, responseNumber);
+        this.rule = rule;
+        this.executedAction = executedAction;
+        this.triggerer = triggerer;
+        this.ruleTriggerer = ruleTriggerer;
+        this.channel = channel;
+        this.messageId = messageId;
+        this.alertSystemMessageId = alertSystemMessageId;
+        this.content = content;
+        this.matchedKeyword = matchedKeyword;
+        this.matchedContent = matchedContent;
+    }
+
+    /**
+     * The {@link AutoModerationRule} that the action belongs to.
+     *
+     * @return The rule.
+     */
+    @Nonnull
+    public AutoModerationRule getRule()
+    {
+        return rule;
+    }
+
+    /**
+     * The {@link AutoModerationAction} that was executed when the rule was triggered.
+     *
+     * @return the executed action.
+     */
+    @Nonnull
+    public AutoModerationAction getExecutedAction()
+    {
+        return executedAction;
+    }
+
+    /**
+     * The {@link User} that caused the rule to be triggered.
+     *
+     * @return The user.
+     */
+    @Nonnull
+    public User getTriggerer()
+    {
+        return triggerer;
+    }
+
+    /**
+     * The {@link TriggerType} that caused the rule to be triggered.
+     *
+     * @return The trigger type.
+     */
+    @Nonnull
+    public TriggerType getTriggerType()
+    {
+        return ruleTriggerer;
+    }
+
+    /**
+     * The {@link GuildChannel} where the rule was triggered.
+     *
+     * @return The channel.
+     */
+    @Nullable
+    public GuildChannel getChannel()
+    {
+        return channel;
+    }
+
+    /**
+     * The message Id of the message that triggered the rule.
```

Comment:
```suggestion
     * Returns the id of the message that triggered the rule.
```

Consider linking message as well.